src/app/community-view/community-view.component.spec.ts

```diff
@@ -0,0 +1,136 @@
+import { async, ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+
+import { CommunityViewComponent } from './community-view.component';
+import { AppMaterialModule } from '../app-material.module';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatMomentDateModule } from '@angular/material-moment-adapter';
+import { ReactiveFormsModule } from '@angular/forms';
+import { UserIdentity } from '../shared/user-identity';
+import { CommunityResponse } from '../communities/community-response';
+import { CommunityType } from '../communities/community-type.enum';
+import { User } from '../users/user';
+import { UserIdentityService } from '../shared/user-identity.service';
+import { of } from 'rxjs';
+import { GlobalErrorHandlerService } from '../error/global-error-handler.service';
+import { CommunitiesService } from '../communities/communities.service';
+import { RouterTestingModule } from '@angular/router/testing';
+import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
+import { DeleteConfirmationDialogComponent } from '../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+
+const authenticatedUser: UserIdentity = {
+  userId: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+  userName: 'tester',
+  firstName: 'Toni',
+  lastName: 'Tester',
+  email: 'toni.tester@myskills.io',
+  roles: ['ROLE_USER']
+};
+
+const community: CommunityResponse = {
+  id: '123',
+  title: 'group1',
+  description: 'super group description',
+  type: CommunityType.OPENED,
+  links: [
+    {
+      name: 'google',
+      href: 'https://www.google.com'
+    },
+    {
+      name: 'stackoveflow',
+      href: 'https://stackoverflow.com/'
+    }],
+  members: [
+    {
+      id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17666',
+      userName: 'tester'
+    } as User
+  ],
+  managers: [
+    {
+      id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17666',
+      userName: 'tester'
+    } as User
+  ]
+};
+
+describe('CommunityViewComponent', () => {
+  let component: CommunityViewComponent;
+  let fixture: ComponentFixture<CommunityViewComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        AppMaterialModule,
+        BrowserAnimationsModule,
+        MatMomentDateModule,
+        ReactiveFormsModule,
+        RouterTestingModule
+      ],
+      declarations: [CommunityViewComponent, DeleteConfirmationDialogComponent],
+      providers: [
+        {
+          provide: CommunitiesService, useValue: jasmine.createSpyObj('communityService', {
+            'getCommunity': of(community),
+            'leaveCommunity': of<void>(),
```

Comment:
There should be `CommunityResponse` instead of `void`.

Reply:
done